### DISCUSSION TITLE
support of tailwind v4 ?

### DISCUSSION THREAD

**Discussion by flozero** [dcastil/tailwind-merge#468] [Ideas] [answered]
Hello thank you again for this awesome work !

I was wondering if your library would work out of the box with tailwind v4 ? As some of the new library classes have breaking changes. I know you have a really extensionnable lib. But wanted to check the state of it

I would love to see this library integrated to tailwind itself as its just so useful but I guess it wont happen soon.

**Comment by dcastil** [comment, marked as answer]
Hey @flozero! 👋

Thanks for your patience, just came back from a vacation. tailwind-merge will support Tailwind CSS v4 once it gets released, but probably not in the current version. I usually release a new tailwind-merge version with support for a major Tailwind CSS reelase it within a couple days.

Depending on the Tailwind features, there might be breaking changes necessary in tailwind-merge in which case I'd release Tailwind CSS support under tailwind-merge@v3.

**Reply by flozero** [reply]
That's great to hear ! And that's totally fair as it will require a lot of works depending no how they are going to structure their API

Thank you for your response !

**Comment by bdrtsky** [comment]
Hello, @dcastil what's your prognosis about Tailwind 4 support (when it will be released)? Did you tried it? Wondering about this, since wanted to adopt Tailwind 4, but without merge it doesn't make any sense. Thank you.

**Reply by dcastil** [reply]
Hey @bdrtsky! 👋

My plan is to release a new (most probably major) version of tailwind-merge a couple days after tailwindcss v4 gets released. I want to do this after the stable release mostly because I don't have the capacity to shadow the tailwindcss beta releases and because that would increase the likelihood of bugs in tailwind-merge.

I didn't try tailwindcss v4 out yet but I've read the blog post at https://tailwindcss.com/docs/v4-beta. It looks like adopting those changes will not be too hard for tailwind-merge. I expect that I can get support for most of the stuff just by tweaking the default config of tailwind-merge.

**Reply by dcastil** [reply]
If you want to try out the tailwindcss v4 beta, you can probably do it with the current tailwind-merge version and it will mostly already work. Only be cautious with passing classes that have completely new syntax into tailwind-merge.

**Reply by bdrtsky** [reply]
Thank you for your response @dcastil

I was worried since they using Lightning CSS instead of PostCSS, and wasn't sure if this affects tailwind-merge. Because, honestly, without merge, Tailwind will be pretty much unusable for my case 😁 heavily relying on shadcn-like components styling from parent className.

And thank you for all your work!

**Reply by dcastil** [reply]
Ah yeah tailwind-merge is basically only directly dependent on the Tailwind CSS syntax and nothing else, so Tailwind CSS switching its tech stack underneath doesn't affect tailwind-merge in any way.

Thanks for your nice words!

**Comment by dcastil** [comment]
I created an issue to track progress on Tailwind CSS v4 support here: https://github.com/dcastil/tailwind-merge/issues/513
